server.js

```diff
@@ -1,7 +1,27 @@
-const app = require("./src/app")
+const RabbitMq = require("./src/messageQueue/setupRabbitMq")
 require('dotenv').config()
-PORT = process.env.PORT || 3055
+const {initializeDatabase, releaseDatabaseConnection} = require('./src/dbs/setupDatabase')
+const initializeRabbitMQ = require('./src/messageQueue/setupRabbitMq')
+const {initializeWebServer} = require('./src/app')
+// Using Dependence Injection For DB and Rabbit Mq for easily testing
+async function start() {
+        console.time(">>> Start WebServer")
+        await initializeDatabase()
+        try {
+          await initializeWebServer();
+          await initializeRabbitMQ(RabbitMq)
+        } catch (error) {
+          releaseDatabaseConnection()
```

Comment:
I think should move the logic releaseDatabaseConnection to after the catch at line 24, **and check if already has the connection or not if not release for more better.**

Reply:
Oke I understand I will change it in the next commit